File: Programmers/연습문제/Level 1/[1차] 비밀지도/Solution.js
function solution(n, arr1, arr2) {
    const answer = getDecodedMap(n, arr1, arr2);
    return answer;
}

function getBinaryArray(n, arr) {
    let binaryArr = new Array();
    for (const num of arr) {
        const row = Array.from(num.toString(2)).map(a => parseInt(a));
        const L = row.length;
        for (let i=0; i<n-L; i++) {
            row.unshift(0);
        }
        binaryArr.push(row);
    }
    return binaryArr;
}

function getDecodedMap(n, arr1, arr2) {
    const binaryArr1 = getBinaryArray(n, arr1);
    const binaryArr2 = getBinaryArray(n, arr2);

    let decodedMap = new Array(n);
    for (let i=0; i<n; i++) {
        decodedMap[i] = new Array(n);
    }

    for (let r=0; r<n; r++) {
        for (let c=0; c<n; c++) {
            decodedMap[r][c] = binaryArr1[r][c] || binaryArr2[r][c] ? '#' : ' ';
        }
    }

    for (let i=0; i<n; i++) {
        decodedMap[i] = decodedMap[i].join('');
    }

    return decodedMap;
}


// TC
const n = 5;
const arr1 = [9, 20, 28, 18, 11];
const arr2 = [30, 1, 21, 17, 28];
console.log(solution(n, arr1, arr2)); // ["#####","# # #", "### #", "# ##", "#####"]

File: Programmers/연습문제/Level 1/[1차] 비밀지도/Solution.swift
func solution(_ n:Int, _ arr1:[Int], _ arr2:[Int]) -> [String] {
    return (0..<n).map {
        let row = String(arr1[$0] | arr2[$0], radix: 2)
        let padding = String(repeating: "0", count: n - row.count )
        let paddedRow = padding + row
        return paddedRow.reduce("") { $0 + ($1 == "1" ? "#" : " ") }
    }
}

    
// TC
let n = 5
let arr1 = [9, 20, 28, 18, 11]
let arr2 = [30, 1, 21, 17, 28]
print(solution(n, arr1, arr2))
